confd crash if the template error

Hi,

When I have a char '$' in the template, then the confd crash.
the version is 0.9.0

here is the log

```
panic: template: haproxy.cfg.tmpl:21: bad character U+0024 '$'

goroutine 14 [running]:
text/template.Must(0x0, 0x7f0882a63940, 0xc2080b2c20, 0x1)
 /usr/local/go/src/text/template/helper.go:23 +0x50
github.com/kelseyhightower/confd/resource/template.(*TemplateResource).createStageFile(0xc208072a00, 0x0, 0x0)
 /Users/bacongobbler/go/src/github.com/kelseyhightower/confd/resource/template/resource.go:119 +0x550
github.com/kelseyhightower/confd/resource/template.(*TemplateResource).process(0xc208072a00, 0x0, 0x0)
 /Users/bacongobbler/go/src/github.com/kelseyhightower/confd/resource/template/resource.go:245 +0xa8
github.com/kelseyhightower/confd/resource/template.process(0xc2080403e0, 0x1, 0x1, 0x0, 0x0)
 /Users/bacongobbler/go/src/github.com/kelseyhightower/confd/resource/template/processor.go:26 +0x99
github.com/kelseyhightower/confd/resource/template.(*intervalProcessor).Process(0xc208064700)
 /Users/bacongobbler/go/src/github.com/kelseyhightower/confd/resource/template/processor.go:53 +0xf5
created by main.main
 /Users/bacongobbler/go/src/github.com/kelseyhightower/confd/confd.go:46 +0x58b

goroutine 1 [runnable]:
main.main()
 /Users/bacongobbler/go/src/github.com/kelseyhightower/confd/confd.go:50 +0x9d4

goroutine 5 [syscall, 44 minutes]:
os/signal.loop()
 /usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
 /usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 6 [select, 44 minutes]:
net.func·019()
 /usr/local/go/src/net/dnsclient_unix.go:241 +0x2f4
created by net.loadConfig
 /usr/local/go/src/net/dnsclient_unix.go:269 +0x20c

goroutine 108 [runnable]:
text/template/parse.(*lexer).errorf(0xc208064b80, 0x7f5dd0, 0x11, 0xc20806bf50, 0x1, 0x1, 0xc200000000)
 /usr/local/go/src/text/template/parse/lex.go:165 +0xc0
text/template/parse.lexFieldOrVariable(0xc208064b80, 0x7, 0x1)
 /usr/local/go/src/text/template/parse/lex.go:408 +0x32b
text/template/parse.lexField(0xc208064b80, 0x876758)
 /usr/local/go/src/text/template/parse/lex.go:375 +0x31
text/template/parse.(*lexer).run(0xc208064b80)
 /usr/local/go/src/text/template/parse/lex.go:198 +0x5d
created by text/template/parse.lex
 /usr/local/go/src/text/template/parse/lex.go:191 +0x1ac

goroutine 16 [IO wait]:
net.(*pollDesc).Wait(0xc208010990, 0x72, 0x0, 0x0)
 /usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208010990, 0x0, 0x0)
 /usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc208010930, 0xc208096000, 0x1000, 0x1000, 0x0, 0x7f0882a67be0, 0xc2080b29d0)
 /usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc208040448, 0xc208096000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
 /usr/local/go/src/net/net.go:121 +0xdc
net/http.noteEOFReader.Read(0x7f0882a69678, 0xc208040448, 0xc20804e478, 0xc208096000, 0x1000, 0x1000, 0x709260, 0x0, 0x0)
 /usr/local/go/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc2080ab420, 0xc208096000, 0x1000, 0x1000, 0xc208012000, 0x0, 0x0)
 <autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc20804c7e0)
 /usr/local/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Peek(0xc20804c7e0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
 /usr/local/go/src/bufio/bufio.go:132 +0xf0
net/http.(*persistConn).readLoop(0xc20804e420)
 /usr/local/go/src/net/http/transport.go:842 +0xa4
created by net/http.(*Transport).dialConn
 /usr/local/go/src/net/http/transport.go:660 +0xc9f

goroutine 17 [select]:
net/http.(*persistConn).writeLoop(0xc20804e420)
 /usr/local/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
 /usr/local/go/src/net/http/transport.go:661 +0xcbc

```

--- a/resource/template/resource.go
+++ b/resource/template/resource.go
@@ -116,7 +116,11 @@ func (t *TemplateResource) createStageFile() error {
 	}
 	defer temp.Close()
 	log.Debug("Compiling source template " + t.Src)
-	tmpl := template.Must(template.New(path.Base(t.Src)).Funcs(t.funcMap).ParseFiles(t.Src))
+	tmpl, err := template.New(path.Base(t.Src)).Funcs(t.funcMap).ParseFiles(t.Src)
+	if err != nil {
+		return fmt.Errorf("Unable to process template %s, %s", t.Src, err)
+	}
+
 	if err = tmpl.Execute(temp, nil); err != nil {
 		return err
 	}

